x-pack/plugins/screenshotting/server/browsers/chromium/driver_factory/index.ts

```diff
@@ -180,10 +193,9 @@ export class HeadlessChromiumDriverFactory {
       };
       const { terminate$ } = safeChildProcess(logger, childProcess);
 
-      // this is adding unsubscribe logic to our observer
-      // so that if our observer unsubscribes, we terminate our child-process
+      // Ensure that the browser is closed once the observable completes.
       observer.add(() => {
-        logger.debug(`The browser process observer has unsubscribed. Closing the browser...`);
+        logger.debug(`It looks like the browser is no longer being used. Closing the browser...`);
```

Comment:
This happens even if we've explicitly closed the browser, right? Maybe we should make this conditional:
```suggestion
        if (!page.isClosed()) {
          logger.debug(`It looks like the browser is no longer being used. Closing the browser...`);
          childProcess.kill(); // ignore async
        }
```

Reply:
Good idea!

Reply:
I think we should only have this guard in place for the `kill()` function since we'll have to repeat this check otherwise.

I'll add a test to describe the close behaviour we want!

Reply:
Ah, I see we want to avoid logging unnecessarily. Will add the check.